pkg/compose/watch.go

```diff
@@ -353,7 +353,7 @@ func loadDevelopmentConfig(service types.ServiceConfig, project *types.Project)
 
 func checkIfPathAlreadyBindMounted(watchPath string, volumes []types.ServiceVolumeConfig) bool {
 	for _, volume := range volumes {
-		if volume.Bind != nil && strings.HasPrefix(watchPath, volume.Source) {
+		if volume.Bind != nil && (strings.HasPrefix(watchPath, volume.Source + string(os.PathSeparator)) || watchPath == volume.Source) {
```

Comment:
This won't work if bind mount / watch path applies to a single file:

```yml
services:
   test:
     volumes:
        ./package.json:/app/packags.json
     develop:
      watch:
        - path: package.json
```

Reply:
you could use `filepath.Rel(volume.Source, watchPath)`. If function doesn't return error, then source is actually a parent for watchPath

Reply:
right, please use the `filepath.Rel` approach, reading implementation code it seems to cover a bunch of corner cases

Reply:
> right, please use the `filepath.Rel` approach, reading implementation code it seems to cover a bunch of corner cases

Just added a commit to use `filepath.Rel`!